File: src/entities/enemies/NoslowEnemy.ts
import { Enemy } from './Enemy';
import { NOSLOW_BLOCK_COUNT } from '../../game/constants';


export class NoslowEnemy extends Enemy {
    private blocksRemaining: number = NOSLOW_BLOCK_COUNT;

    getSpriteName(): string {
        return 'vanguard';
    }

    // Noslow uses a different health scaling: 120 + 100*level (not 120 + 120*level)
    protected initStats(): void {
        this.goldMultiplier = 1.5;
        this.maxHealth = 120 + 100 * this.healthLevel;
        this.health = this.maxHealth;
        this.maxSpeed = 65;
        this.speed = this.maxSpeed;
    }

    // Override takeDamage - blocks consume attacks
    takeDamage(amount: number): boolean {
        if (this.blocksRemaining > 0) {
            this.blocksRemaining--;
            return false; // Blocked, not dead
        }
        return super.takeDamage(amount);
    }

    // Override applyPoison - blocks consume poison attempts
    applyPoison(ticks: number): void {
        if (this.blocksRemaining > 0) {
            this.blocksRemaining--;
            return; // Blocked
        }
        super.applyPoison(ticks);
    }

    // Override applySlow - NEVER apply slow effect (permanently immune to slowing)
    applySlow(_amount: number, _duration: number): void {
        // NoslowEnemy is completely immune to slow effects
        // The slow portion is always ignored (speed reduction never applied)
    }

}

File: original-java-game/src/enemies/NoslowEnemy.java
package enemies;

import source.Board;

public class NoslowEnemy extends Enemy {

	int blocks = 3;
	
	public NoslowEnemy(Board gui, int xs, int ys, int nivå) {
		super(gui.getBilder().getVanguard(), gui, xs, ys, gui.getLevel(), nivå);
	}
	
	@Override
	protected void setStats() { 
		maxhealth = 120 + 100 * healthlevel;
		maxspeed = 1.6;
		damage = 1;
	}
	public void slow(double amount, int dmg) {
		//haha cantslow
		if(blocks > 0){
			blocks--;
		}else{
			health -= dmg;
			if(isDead()){
				die();
			}
		}
	}
	public void takedamge(int dmg){
		if(blocks > 0){
			blocks--;
		}else{
			super.takedamge(dmg);
		}
	}
	public void poison(int ticks){
		if(blocks > 0){
			blocks--;
		}else{
			super.poison(ticks);
		}
	}

}//END
